[Jewel Jace Lim] iP

Jeweljace opened the pull request.

Commit:
ip level 0

Commit:
ip level 1
gave helio some personality
added echo functionality and bye command

Commit:
ip level 2 task

added listing functionality

Commit:
ip level 3

Added Task class to manage list
Added "mark" and "unmark" functionalities
mark and unmark command has to be in this format: "mark/unmark<blank><task number>"

Commit:
Tweak the code to comply with a coding standard

Commit:
Added Todo.java, Deadline.java and Event.java

Updated Helio.java to accept new commands: todo, deadline and event

Updated Helio.java for error handling

Commit:
Added Automated Text UI Testing

Use the input/output redirection technique to semi-automate the testing of Helio.

Added input.txt and EXPECTED.txt.

Added fix to Helio.java so that scanner can work with input.txt properly.

gbinw128 commented on src/main/java/Helio.java:
```diff
@@ -0,0 +1,151 @@
+                if (sections[0].isEmpty() || sections[1].isEmpty() || sections[2].isEmpty()) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println(" Meow! Please fill in a description and date/time of start and end! =.=");
+                    System.out.println("____________________________________________________________");
```
Maybe it would be better to replace all the if-else case with different functions or a function with parameters?

gbinw128 commented on src/main/java/Helio.java:
```diff
@@ -0,0 +1,151 @@
+        String logo = "        ╱|\n" +
+                "       (˚ˎ 。7\n" +
+                "        |、˜〵\n" +
+                "       じしˍ,)ノ\n";
```
cute cat!

gbinw128 commented on src/main/java/Helio.java:
```diff
@@ -0,0 +1,151 @@
+                System.out.println("   " + userList[count - 1]);
+                System.out.println(" Now you have " + count + " tasks in the list.");
+                System.out.println("____________________________________________________________");
+            } else if (input.startsWith("deadline ")) {
```
All "else" is in line with the close bracket, nice!

gbinw128 commented on src/main/java/helio/task/Task.java:
```diff
@@ -0,0 +1,26 @@
+    public void markAsNotDone() {
+        isDone = false;
+    }
+
```
Nicely named functions

Amanda-HUANG88 commented:
Very thorough consideration！Maybe it is better to do some refactoring to reduce redudant codes and extract methods.

AWeiBin commented on src/main/java/Helio.java:
```diff
@@ -0,0 +1,151 @@
+    public static void main(String[] args) {
+        Task[] userList = new Task[100];
+        int count = 0;
+        String logo = "        ╱|\n" +
```
Since the logo is a constant, I suggest that it can be a final class level constant

gbinw128 commented on src/main/java/Helio.java:
```diff
@@ -0,0 +1,151 @@
+                "____________________________________________________________");
+
+        Scanner in = new Scanner(System.in);
+        while (in.hasNextLine()) {
```
Entire loop looks like an arrowhead code

AWeiBin commented on src/main/java/Helio.java:
```diff
@@ -0,0 +1,151 @@
+                        " Bye. Hope to see you again soon!\n" +
+                        "____________________________________________________________");
+                break;
+            } else if (input.equals("help")) {
```
Good Job on making a help command

gbinw128 reviewed (commented):
Overall, nice!

AWeiBin commented on src/main/java/Helio.java:
```diff
@@ -0,0 +1,151 @@
+
+public class Helio {
+    public static void main(String[] args) {
+        Task[] userList = new Task[100];
```
Perhaps the variable name can be more specific.

AWeiBin commented on src/main/java/Helio.java:
```diff
@@ -0,0 +1,151 @@
+                "       (˚ˎ 。7\n" +
+                "        |、˜〵\n" +
+                "       じしˍ,)ノ\n";
+        System.out.println("____________________________________________________________\n" +
```
Consider making a function to print the borders

Commit:
Moved custom messages and error handling to separate classes.

Commit:
Modified some of the custom messages for more personality.

Commit:
Modified personality message

roshnidaksha commented on src/main/java/helio/ui/Handle.java:
```diff
@@ -0,0 +1,42 @@
+public final class Handle {
+    private Handle() {
+    }
```
If a constructor is not required for this class, you can remove these lines.

roshnidaksha commented on src/main/java/helio/ui/Handle.java:
```diff
@@ -0,0 +1,42 @@
+
+    private static final String LINE = "____________________________________________________________";
+
+    public static void invalidCommand() {
```
Method names should be a verb. Can you modify it to sound like a verb?

roshnidaksha commented on src/main/java/Helio.java:
```diff
@@ -0,0 +1,112 @@
+            } else if (input.equals("help")) {
+                Message.help();
+            } else if (input.equals("list")) {
+                if (count == 0) {
+                    Message.emptyList();
+                } else {
+                    Message.listHeader();
+                    for (int i = 0; i < count; i++) {
+                        System.out.println((i + 1) + ". " + userList[i]);
+                    }
+                    Message.listFooter();
+                }
```
Implementation details of these features can be extracted out to a new method to improve SLAP.
The same comment goes for your other features as well.

roshnidaksha commented on src/main/java/helio/Helio.java:
```diff
@@ -0,0 +1,112 @@
+                    Message.listFooter();
+                }
+            } else if (input.startsWith("mark ")) {
+                String check = input.substring(5).trim();
```
Consider creating a new method for low-level string operations too.

roshnidaksha commented on src/main/java/helio/ui/Handle.java:
```diff
@@ -0,0 +1,42 @@
+public final class Handle {
```
Does this need to be inside `Message.java`? Both the classes only print messages.

roshnidaksha reviewed (commented):
The code is neat generally. 
Some methods can be named better.
You can consider improving SLAP of overall code as well.

Commit:
Made further changes to Helio's messages

Commit:
Added delete functionality to Helio

Changed array implementation to array list

Added further error handling for deadline and event

Commit:
Added storage.java for save function

Tweaked code to enable save to file functionality

Commit:
Merge branch 'branch-Level-7'

Resolved conflicts in Helio.java

Commit:
Fixed errors due to change from Array to java collections ArrayList

Commit:
Packaged the App as a JAR file

Commit:
Fixed error where deadline and delete commands were not writing to saved file

Commit:
Divided classes into suitable packages

Commit:
Merge branch 'branch-A-Packages'

Commit:
Added TaskList.java in task package to contain the task list

Commit:
Added Ui.java to handle all user interactions and also removed Handle.java and Message.java

Commit:
Add Parser.java and *Command classes

Commit:
Update Helio.java

Helio.java was updated to make it compatible with the implemented OOP design

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Update AddTodoCommand.java

Commit:
Update AddEventCommand.java

Commit:
Update AddEventCommand.java

Commit:
Add DateTimeUtil.java

Commit:
Update Parser.java and Storage.java

Commit:
Update Deadline.java and Event.java and Task.java

Commit:
Update AddDeadlineCommand.java and AddEventCommand.java and AddTodoCommand.java

Commit:
Add OnDateCommand.java

Commit:
Update Helio.java

Chatbot now understands dates and times for deadlines and events

Commit:
Merge branch 'branch-Level-8'

Commit:
Add FindCommand.java

Commit:
Update Ui.java and TaskList.java

Commit:
Update Parser.java

Commit:
Merge branch 'branch-Level-9'

Commit:
Add JavaDoc comments to all classes and non-trivial methods

Commit:
Update Helio.java

Commit:
Update README.md

Updated README.md to include User Guide

Commit:
Update Helio.java

Made changes to JavaDoc comments to make it more detailed

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Update MANIFEST.MF

Commit:
Merge branch 'Week-6-Jar'

Commit:
Update OnDateCommand.java

Commit:
Update Ui.java

Changed Helio logo to be compatible with both linux and windows

Commit:
Update Ui.java and Helio.java

Helio bye message was updated to reflect more personality

Jeweljace committed:
Merge pull request #1 from Jeweljace/branch-Merge-PR

Update Ui.java and Helio.java
